src/lib.rs

```diff
@@ -189,21 +193,55 @@ impl Stream for Incoming {
     }      
 }
 
-/// IPC stream of client connection
-pub struct IpcStream {
+/// IPC Connection
+pub struct IpcConnection {
     #[cfg(windows)]
     inner: tokio_named_pipes::NamedPipe,
     #[cfg(not(windows))]
     inner: tokio_uds::UnixStream,
 }
 
-impl Read for IpcStream {
+#[deprecated(since="0.1.5", note = "Please use `IpcConnection` instead")]
+pub type IpcStream = IpcConnection;
+
+
+impl IpcConnection {
+    pub fn connect<P: AsRef<Path>>(path: P, handle: &Handle) -> io::Result<IpcConnection> {
+        Ok(IpcConnection{
+            inner: Self::_connect(path.as_ref(), handle)?,
+        })
+    }
+
+    #[cfg(unix)]
+    pub fn _connect(path: &Path, handle: &Handle) -> io::Result<tokio_uds::UnixStream> {
+        tokio_uds::UnixStream::connect(&path, &handle)
+    }
+
+    #[cfg(windows)]
+    pub fn _connect(path: &Path, handle: &Handle) -> io::Result<NamedPipe> {
```

Comment:
why underscore? does it need to be public also?

Reply:
It doesn't need to be public. Renamed `_connect` to `inner` as well.

Reply:
maybe `connect_internal`, sorry? 
`inner` is used for fields mostly... :)

Reply:
Sure, `connect_internal` it will be then :)